createFeatureFlag/main.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/Real-Dev-Squad/feature-flag-backend/database"
+	"github.com/Real-Dev-Squad/feature-flag-backend/models"
+	"github.com/Real-Dev-Squad/feature-flag-backend/utils"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/go-playground/validator/v10"
+	"github.com/google/uuid"
+	"log"
+	"net/http"
+	"time"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+}
+
+func createFeatureFlag(db *dynamodb.DynamoDB, createFeatureFlagRequest utils.CreateFeatureFlagRequest) error {
+	featureFlag := models.FeatureFlag{
+		Id:          uuid.New().String(),
+		Name:        createFeatureFlagRequest.FlagName,
+		Description: createFeatureFlagRequest.Description,
+		CreatedBy:   createFeatureFlagRequest.UserId,
+		UpdatedBy:   createFeatureFlagRequest.UserId,
+		CreatedAt:   time.Now().Unix(),
+		UpdatedAt:   time.Now().Unix(),
+		Status:      utils.ENABLED,
+	}
+
+	//marshal to ddb attribute value
+	item, err := dynamodbattribute.MarshalMap(featureFlag)
+	if err != nil {
+		log.Println("Error marshalling object to DynamoDB AttributeValue: %v", err)
+		return err
+	}
+
+	log.Println("item is", item)
+	input := &dynamodb.PutItemInput{
+		TableName: aws.String(database.GetFeatureFlagTableName()),
+		Item:      item,
+	}
+
+	log.Println(db, " is the db object")
+	_, err = db.PutItem(input)
+	if err != nil {
+		log.Println("Error putting item to Dynamodb: %v", err)
+		return err
+	}
+	return nil
+}
+
+// //TODO: use the validation Error struct response and send it to the user
+
+func handleValidationError(err error) []utils.ValidationError {
+	var errors []utils.ValidationError
+
+	for _, err := range err.(validator.ValidationErrors) {
+		errors = append(errors, utils.ValidationError{
+			Field: err.Field(),
+			Error: err.Tag(),
+		})
+	}
+	return errors
```

Comment:
do we need an array for this? If any validation fails just send that one.

Reply:
They can be multiple validation errors possible, say first and second is missing, then we get two errors right, need to use them to show to the user so have written a TODO, tried doing wasn't successful. @RitikJaiswal75

Reply:
Once any error comes stop the lambda and return the error what are we waiting for all the errors to accumulate and then send it?

Lambdas spin up for a very short duration we should try to keep the code fast for such cases.

Reply:
I agree with @RitikJaiswal75. Also, going forward I think it will be better to remove the validation logic from lambda function and add validate request in API Gateway directly
Reference - https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-method-request-validation.html

Reply:
> Once any error comes stop the lambda and return the error what are we waiting for all the errors to accumulate and then send it?

> Lambdas spin up for a very short duration we should try to keep the code fast for such cases.

Agreed

Reply:
I'm thinking to have it like 
```
if err.(Validator.ValidationErrors) != nil {
                        return events.APIGatewayProxyResponse{
				Body:       "Check the request body passed name, description and userId are required.",
				StatusCode: http.StatusBadRequest,
			}, nil
			
	}
```			
are you inline @RitikJaiswal75 ?

Reply:
its basically for line 66 where you are pushing all the errors as soon as a single error comes in we can directly send it back and stop the lambda. rather than waiting for all the errors to pile up and pushing them into a slice. this will also reduce the amount of time we consume over a lambda.

Reply:
```
func handleValidationError(err error) err {

	for _, err := range err.(validator.ValidationErrors) {
		return err
	}
	return nil

}
```
this way. @RitikJaiswal75

Reply:
yes, this looks good. Now your loop will return an error as soon as one is encountered and halt itself consuming less time over lambda.

Reply:
@RitikJaiswal75 @shubham-y please check updated